engine/src/test/java/org/hibernate/validator/test/internal/engine/ValidatorTest.java

```diff
@@ -325,6 +344,20 @@ public String getFoo() {
 		}
 	}
 
+	class I {
+		@Valid
+		Map<String, J> map = new HashMap<String, J>();
```

Comment:
Thanks for the explanation ! 

Everything else looks good to me, but I'm not as familiar with this part of code as any of you are :) so I'm probably not the one who should merge it in :)